tox.ini

```diff
@@ -0,0 +1,136 @@
+[tox]
+envlist = py{34,35,36,37}-{all,pip,clang,npm,gem,go,perl,php,cabal,java,java7,java8,scala,elm,r,dart,julia,lua,infer,opam,apt,adhoc,disabled,win,swift,mono}-{list,check,collectonly,skip,noskip}-codecov
+minversion = 3.4
+
+[travis:env]
+TRAVIS =
+  true: codecov
+TRAVIS_LANGUAGE =
+  python: pip-noskip
+  node_js: py36-npm-noskip
+  ruby: py36-gem-noskip
+  haskell: py36-cabal-noskip
+  go: py36-go-noskip
+  perl: py36-perl-noskip
+  php: py36-php-noskip
+  scala: py36-scala-noskip
+  elm: py36-elm-noskip
+  r: py36-r-noskip
+  dart: py36-dart-noskip
+  julia: py36-julia-noskip
+  objective_c: py36-swift-noskip
+  swift: py36-swift-noskip
+  objectivec: py36-swift-noskip
+  objective-c: py36-swift-noskip
+  csharp: py36-mono-noskip
+  minimal,generic: py36-adhoc-noskip
+TRAVIS_JDK_VERSION =
+  oraclejdk11: py36-java-skip
+  oraclejdk9: py36-java-skip
+  oraclejdk8: py36-java8-noskip
+  openjdk11: py36-java-skip
+  openjdk10: py36-java-skip
+  openjdk9: py36-java-skip
+  openjdk8: py36-java8-noskip
+  openjdk7: py36-java7-noskip
+# apt_get is the only group allowed to skip
+BEARS =
+  apt_get: py36-apt-skip
+  lua: py36-lua-noskip
+  infer: py36-infer-noskip
+  opam: py36-opam-noskip
+  adhoc: py36-adhoc-noskip
+  clang: py36-clang-noskip
+  disabled: py36-disabled-noskip
+
+# pyyaml literally here to support get_tests.py
+[testenv]
+passenv =
+  HOME
+  PATH
+  CI CI_*
+  TRAVIS TRAVIS_*
+  APPVEYOR APPVEYOR_*
+  TOX_*
+  PIP_*
+  VIRTUALENV_*
+  LOCALAPPDATA
+  GEM_HOME
+  GEM_PATH
+  JULIA_PROJECT
+  GOROOT
+  GOPATH
+  BEARS
+  BEAR_LIST
+  DISABLE_BEARS
+  R_LIBS_USER
+  R_LIBS_SITE
+  _R_CHECK_CRAN_INCOMING_
+  NOT_CRAN
+  R_PROFILE
```

Comment:
two or four space indents ; choose one!

We are using two space indents for setup.cfg , which is also an .ini , and is a template for all repos, so that is the more sensible choice.